## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [1]:
!ls

CONTRIBUTING.md
LICENSE.md
README.md
awesome.gif
data.sqlite
im.db
movie_data_erd.jpeg
student.ipynb
zippedData


# Importing the relevant libraries
Begin by importing pandas, numpy, csv with their aliases

In [2]:
# Your code here - remember to use markdown cells for comments as well!
#importing libraries
import pandas as pd
import sqlite3
import csv
import numpy as np
from datetime import datetime


# Exploring the datasets
Preview the first dataset, which is from "tn.movie_budgets.csv", which is inside the zipped data

In [3]:



#Load the data from tn.movie_budgets.csv into a dataframe
movies_budget = pd.read_csv('zippedData/tn.movie_budgets.csv')

#Cast to numeric data type
movies_budget['production_budget'] = movies_budget['production_budget'].replace('[^.0-9]', '', regex=True).astype('float32')
movies_budget['domestic_gross'] = movies_budget['domestic_gross'].replace('[^.0-9]', '', regex=True).astype('float32')
movies_budget['worldwide_gross'] = movies_budget['worldwide_gross'].replace('[^.0-9]', '',regex = True).astype('float64')
movies_budget.head(5)

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0        425000000.0     760507648.0     2.776345e+09  
1        410600000.0     241063872.0     1.045664e+09  
2        350000000.0      42762352.0     1.497624e+08  
3        330600000.0     459005856.0     1.403014e+09  
4        317000000.0     620181376.0     1.316722e+09

The production budget, domestic gross, and worldwide gross are then cast to float type for mathematical manipulation.

In [4]:
movies_budget.shape

(5782, 6)

In [5]:
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5782 non-null   int64  
 1   release_date       5782 non-null   object 
 2   movie              5782 non-null   object 
 3   production_budget  5782 non-null   float32
 4   domestic_gross     5782 non-null   float32
 5   worldwide_gross    5782 non-null   float64
dtypes: float32(2), float64(1), int64(1), object(2)
memory usage: 226.0+ KB


In [6]:
movies_budget.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

The movie budget Data frame has 5,782 rows and 6 columns namely: id,release_date, movie, production_budget, domestic gross and worldwide_gross. The data types are as float,object and int. The are no null values.

Finding out the movies budget, first we get the international gross by subtracting the domestic gross from the woorldwide gross

In [7]:
#getting the international gross
movies_budget['international_gross'] = movies_budget['worldwide_gross'] - movies_budget['domestic_gross']
cols = list(movies_budget.columns)

movies_budget.head(10)

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
5   6  Dec 18, 2015         Star Wars Ep. VII: The Force Awakens   
6   7  Apr 27, 2018                       Avengers: Infinity War   
7   8  May 24, 2007   Pirates of the Caribbean: At World’s End   
8   9  Nov 17, 2017                               Justice League   
9  10   Nov 6, 2015                                      Spectre   

   production_budget  domestic_gross  worldwide_gross  international_gross  
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09  
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08  
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08  
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08  
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08  
5        306000000.0     936662208.0     2.053311e+09         1.116649e+09  
6        300000000.0     678815488.0     2.048134e+09         1.369319e+09  
7        300000000.0     309420416.0     9.634204e+08         6.540000e+08  
8        300000000.0     229024288.0     6.559452e+08         4.269209e+08  
9        300000000.0     200074176.0     8.796209e+08         6.795467e+08

The release date is an object, in order to filter the years convert to date time type.

In [8]:
#converting object to dateTime
movies_budget['release_date'] = pd.to_datetime(movies_budget['release_date'], errors='coerce')
movies_budget['year'] = movies_budget['release_date'].dt.strftime('%Y')
movies_budget.head(10)

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   
5   6   2015-12-18         Star Wars Ep. VII: The Force Awakens   
6   7   2018-04-27                       Avengers: Infinity War   
7   8   2007-05-24   Pirates of the Caribbean: At World’s End   
8   9   2017-11-17                               Justice League   
9  10   2015-11-06                                      Spectre   

   production_budget  domestic_gross  worldwide_gross  international_gross  \
0        425000000.0     760507648.0     2.776345e+09         2.015838e+09   
1        410600000.0     241063872.0     1.045664e+09         8.046000e+08   
2        350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3        330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4        317000000.0     620181376.0     1.316722e+09         6.965404e+08   
5        306000000.0     936662208.0     2.053311e+09         1.116649e+09   
6        300000000.0     678815488.0     2.048134e+09         1.369319e+09   
7        300000000.0     309420416.0     9.634204e+08         6.540000e+08   
8        300000000.0     229024288.0     6.559452e+08         4.269209e+08   
9        300000000.0     200074176.0     8.796209e+08         6.795467e+08   

   year  
0  2009  
1  2011  
2  2019  
3  2015  
4  2017  
5  2015  
6  2018  
7  2007  
8  2017  
9  2015

In [9]:
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id                   5782 non-null   int64         
 1   release_date         5782 non-null   datetime64[ns]
 2   movie                5782 non-null   object        
 3   production_budget    5782 non-null   float32       
 4   domestic_gross       5782 non-null   float32       
 5   worldwide_gross      5782 non-null   float64       
 6   international_gross  5782 non-null   float64       
 7   year                 5782 non-null   object        
dtypes: datetime64[ns](1), float32(2), float64(2), int64(1), object(2)
memory usage: 316.3+ KB


# Filtering the Data
Here we will drop the rows without the values for the domestic, international and worldwide gross.

In [10]:
movies_budget.sort_values('worldwide_gross')

id release_date                                    movie  \
5037  38   2019-04-23  Living Dark: The Story of Ted the Caver   
3975  76   2015-05-15                           Pound of Flesh   
4627  28   2011-06-28                                     2:13   
4628  29   2013-01-29  Batman: The Dark Knight Returns, Part 2   
3947  48   2019-06-21                           Burn Your Maps   
...   ..          ...                                      ...   
33    34   2015-06-12                           Jurassic World   
6      7   2018-04-27                   Avengers: Infinity War   
5      6   2015-12-18     Star Wars Ep. VII: The Force Awakens   
42    43   1997-12-19                                  Titanic   
0      1   2009-12-18                                   Avatar   

      production_budget  domestic_gross  worldwide_gross  international_gross  \
5037          1750000.0             0.0     0.000000e+00         0.000000e+00   
3975          7500000.0             0.0     0.000000e+00         0.000000e+00   
4627          3500000.0             0.0     0.000000e+00         0.000000e+00   
4628          3500000.0             0.0     0.000000e+00         0.000000e+00   
3947          8000000.0             0.0     0.000000e+00         0.000000e+00   
...                 ...             ...              ...                  ...   
33          215000000.0     652270656.0     1.648855e+09         9.965842e+08   
6           300000000.0     678815488.0     2.048134e+09         1.369319e+09   
5           306000000.0     936662208.0     2.053311e+09         1.116649e+09   
42          200000000.0     659363968.0     2.208208e+09         1.548844e+09   
0           425000000.0     760507648.0     2.776345e+09         2.015838e+09   

      year  
5037  2019  
3975  2015  
4627  2011  
4628  2013  
3947  2019  
...    ...  
33    2015  
6     2018  
5     2015  
42    1997  
0     2009  

[5782 rows x 8 columns]

In [18]:
#drop all all rows without the gross values
movies_budget_drop = movies_budget[(movies_budget['domestic_gross'] == 0) & (movies_budget['worldwide_gross'] == 0)]

In [17]:
movies_budget.drop(movies_budget_drop.index, axis=0, inplace=True)
movies_budget

id release_date                                        movie  \
0      1   2009-12-18                                       Avatar   
1      2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2      3   2019-06-07                                 Dark Phoenix   
3      4   2015-05-01                      Avengers: Age of Ultron   
4      5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   
...   ..          ...                                          ...   
5775  76   2006-05-26                                       Cavite   
5776  77   2004-12-31                              The Mongol King   
5778  79   1999-04-02                                    Following   
5779  80   2005-07-13                Return to the Land of Wonders   
5781  82   2005-08-05                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  international_gross  \
0           425000000.0     760507648.0     2.776345e+09         2.015838e+09   
1           410600000.0     241063872.0     1.045664e+09         8.046000e+08   
2           350000000.0      42762352.0     1.497624e+08         1.070000e+08   
3           330600000.0     459005856.0     1.403014e+09         9.440081e+08   
4           317000000.0     620181376.0     1.316722e+09         6.965404e+08   
...                 ...             ...              ...                  ...   
5775             7000.0         70071.0     7.164400e+04         1.573000e+03   
5776             7000.0           900.0     9.000000e+02         0.000000e+00   
5778             6000.0         48482.0     2.404950e+05         1.920130e+05   
5779             5000.0          1338.0     1.338000e+03         0.000000e+00   
5781             1100.0        181041.0     1.810410e+05         0.000000e+00   

      year  
0     2009  
1     2011  
2     2019  
3     2015  
4     2017  
...    ...  
5775  2006  
5776  2004  
5778  1999  
5779  2005  
5781  2005  

[5415 rows x 8 columns]

In order to have relevant budgeting information ,filter the years to get the data from the last five years.

In [20]:
#convert from object to string
movies_budget['year'] = movies_budget['year'].astype(int)


In [26]:
#drop rows which are not between the years 2017 and 2022
drop_=  movies_budget[(movies_budget['year'] <= 2015)]
movies_budget.drop(drop_.index, axis=0, inplace=True)
movies_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 189 entries, 2 to 5466
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   id                   189 non-null    int64         
 1   release_date         189 non-null    datetime64[ns]
 2   movie                189 non-null    object        
 3   production_budget    189 non-null    float32       
 4   domestic_gross       189 non-null    float32       
 5   worldwide_gross      189 non-null    float64       
 6   international_gross  189 non-null    float64       
 7   year                 189 non-null    int32         
dtypes: datetime64[ns](1), float32(2), float64(2), int32(1), int64(1), object(1)
memory usage: 16.1+ KB


Finding the percentage profits of the movies based on the production budget and the world wide gross.

In [ ]:
movies_budget.describe()

In [ ]:
movies_gross = pd.read_csv('zippedData/bom.movie_gross.csv')
movies_gross.head()

In [ ]:
movies_gross.shape

In [ ]:
movies_gross.info()

The tmdb.movies files will give us information on the popularity of a movie with respect to the language.

In [ ]:
movies_budget['release_date'] =movies_budget.loc[datetime.release_date(year=2017,month=1,day=1):datetime.release_date(year=2022,month=1,day=1)]
movies_budget

In [ ]:
tmd = pd.read_csv('zippedData/tmdb.movies.csv',index_col=0)
tmd.head(5)

In [ ]:
tmd['original_language'].unique

In [ ]:
tmd.shape

In [ ]:
tmd.info()

In [ ]:
tmd['popularity'].max()


In [ ]:
reviews = pd.read_csv('zippedData/rt.reviews.tsv', delimiter='\t',encoding='unicode_escape')
reviews.head(10)

In [ ]:
reviews.info()

In [ ]:
reviews.dropna(axis=0, inplace = True)
reviews.head(10)

From the reviews dataframe it is impossible to get valid data that can be used to draw conclusion on the ratings and the reviews

# checking data from im.db

In [ ]:
conn = sqlite3.connect('zippedData/im.db')
cur = conn.cursor()

# Find the table names in im.db

In [ ]:
cur.execute("""SELECT name FROM sqlite_master WHERE type = 'table';""")
# Fetch the result and store it in table_names
table_names = cur.fetchall()
table_names

Im.db has 8 tables. The movie_ratings and movie_basics are more relevant to our study as they will help gain information on the highly rated genres.

In [ ]:
ratings = pd.read_sql_query("""
SELECT movie_id, start_year, runtime_minutes, genres, averagerating, numvotes
FROM movie_basics
JOIN movie_ratings
    USING(movie_id)
WHERE start_year >= 201
LIMIT 10
;""", conn)

ratings

In [ ]:
movie_akas = pd.read_sql_query("""SELECT * 
FROM movie_akas
;""",conn)

movie_akas.tail(10)

In [ ]:
#find the languages 
movie_akas['language'].unique()
movie_akas['language'].value_counts().head(10)

In [ ]:
movie_info = pd.read_csv('zippedData/rt.movie_info.tsv', delimiter ='\t', encoding='unicode_escape' )
movie_info

In [ ]:
movie_info.info()